demo/run-osm-demo.sh

```diff
@@ -105,6 +105,8 @@ if [ "$CERT_MANAGER" = "vault" ]; then
       --deploy-prometheus="$DEPLOY_PROMETHEUS" \
       --enable-prometheus-scraping="$ENABLE_PROMETHEUS_SCRAPING" \
       --envoy-log-level "$ENVOY_LOG_LEVEL" \
+      --enable-privileged-init-container "$ENABLE_PRIVILEGED_INIT_CONTAINER" \
```

Comment:
If we use `--set` for this chart value (which I think we should), we shouldn't have to expose this flag via `.env` file.